src/sgp/Input.cc

```diff
@@ -19,17 +19,19 @@ static BOOLEAN fCursorWasClipped = FALSE;
 static SGPRect gCursorClipRect;
 
 
-// These data structure are used to track the mouse while polling
-
-static UINT32 guiSingleClickTimer;
-
+// These data structures are used to track the mouse while polling
 static UINT32 guiLeftButtonRepeatTimer;
 static UINT32 guiRightButtonRepeatTimer;
 static UINT32 guiMiddleButtonRepeatTimer;
+static UINT32 guiX1ButtonRepeatTimer;
+static UINT32 guiX2ButtonRepeatTimer;
 
 BOOLEAN gfLeftButtonState;  // TRUE = Pressed, FALSE = Not Pressed
 BOOLEAN gfRightButtonState; // TRUE = Pressed, FALSE = Not Pressed
 BOOLEAN gfMiddleButtonState;// TRUE = Pressed, FALSE = Not Pressed
+BOOLEAN gfX1ButtonState;// TRUE = Pressed, FALSE = Not Pressed
+BOOLEAN gfX2ButtonState;// TRUE = Pressed, FALSE = Not Pressed
+
```

Comment:
There's a lot of duplication the way we handle keys now. Button state could be a joint bitfield (potentially including modifier keys) and the repeat timer could be a single timer.

Reply:
I have reduced duplication for this and also for double click handling. Button state is a `std::bitset` now (without modifier keys, as I haven't touched them in this PR yet). Each button still has its own repeat timer.